### PR TITLE
Fix prow/jenkins lint errors

### PR DESCRIPTION
**Pull request by fejta**
Fixes https://github.com/kubernetes/test-infra/issues/8462

`Succeess` :grin:

/assign @stevekuznetsov @kargakis

### CODE REVIEW

File: prow/jenkins/jenkins.go [hunk 1]
```diff
@@ -197,15 +206,22 @@ type AuthConfig struct {
 	csrfRequestField string
 }
 
+// BasicAuthConfig authenticates with jenkins using user/pass.
 type BasicAuthConfig struct {
 	User  string
 	Token string
 }
 
+// BearerTokenAuthConfig holds the oauth-token.
```
Comment (kargakis) [inline prow/jenkins/jenkins.go]:
```go
// BearerTokenAuthConfig authenticates with jenkins using an oauth token.
```

File: prow/jenkins/jenkins.go [hunk 2]
```diff
@@ -197,15 +206,22 @@ type AuthConfig struct {
...
 type BearerTokenAuthConfig struct {
 	Token string
 }
 
+// NewClient instantiates a client with provided values.
+//
+// logger: creates a standard logger if nil
+// tlsConfig: configures client transport if set, may be nil
+// authConfig: must be set, will create a CrumbRequest if CSRFProtect
```
Comment (kargakis) [inline prow/jenkins/jenkins.go]:
```go
// authConfig: configures the client to connect to Jenkins either via basic auth or oauth and optionally enables csrf protection
```

File: prow/jenkins/jenkins.go (hunk 1 above)
Comment (fejta) [reply prow/jenkins/jenkins.go]: Done

File: prow/jenkins/jenkins.go [hunk 3]
```diff
@@ -197,15 +206,22 @@ type AuthConfig struct {
...
 	Token string
 }
 
+// BearerTokenAuthConfig holds the oauth-token.
 type BearerTokenAuthConfig struct {
 	Token string
 }
 
+// NewClient instantiates a client with provided values.
+//
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/jenkins/jenkins.go]: Done

File: prow/jenkins/jenkins.go (hunk 2 above)
Comment (fejta) [reply prow/jenkins/jenkins.go]: Done

**Pull request merged**